### PR TITLE
Avoid reinstalling injected package without --force

### LINKED ISSUE #1300: inject command does not respect "--force" option
**Describe the bug**

When `--force` is NOT specified, the `pipx inject` command will still install the dependency when it exists.

**How to reproduce**

```bash
pipx install pycowsay
pipx inject pycowsay cowsay # inject cowsay into pycowsay for the first time
pipx inject pycowsay cowsay # inject again without `--force`, but it is still installed again
```

**Expected behavior**

Only install the injected package again when `--force` is specified.

BTW, I can try to submit a pull request.

### PR DESCRIPTION
**Pull request by huxuan**
- [x] I have added a news fragment under `changelog.d/` (if the patch affects the end users)

## Summary of changes

Fix #1300

## Test plan

Tested by running

```
> pipx install pycowsay
  installed package pycowsay 0.0.0.2, installed using Python 3.10.12
  These apps are now globally available
    - pycowsay
  These manual pages are now globally available
    - man6/pycowsay.6
done! ✨ 🌟 ✨
> pipx inject pycowsay cowsay
  injected package cowsay into venv pycowsay
done! ✨ 🌟 ✨
> pipx inject pycowsay cowsay
cowsay already seems to be injected in 'pycowsay'. Not modifying existing installation in '/usr/local/pipx/venvs/pycowsay'. Pass '--force' to force installation.
> pipx inject pycowsay cowsay --force
  injected package cowsay into venv pycowsay
done! ✨ 🌟 ✨
```

### CODE REVIEW

File: changelog.d/1300.bugfix.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Avoid installing existed injected package without --force
```
Comment (chrysle) [inline changelog.d/1300.bugfix.md:1]:
```suggestion
Do not reinstall already injected packages without `--force` being passed.
```

File: src/pipx/commands/inject.py [hunk 2]
```diff
@@ -55,6 +55,19 @@ def inject_dep(
             pip_args=pip_args,
             verbose=verbose,
         )
+
+    if not force and venv.has_package(package_name):
```
Comment (chrysle) [inline src/pipx/commands/inject.py:59]: We might as well test whether `Venv.list_installed_packages()` contains the package. The added function looks fine, though.

File: src/pipx/commands/inject.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 from pipx.commands.common import package_name_from_spec, run_post_install_actions
 from pipx.constants import EXIT_CODE_INJECT_ERROR, EXIT_CODE_OK, ExitCode
 from pipx.emojis import stars
```
Comment (chrysle) [inline src/pipx/commands/inject.py:10]:
```suggestion
from pipx.emojis import hazard, stars
```

File: src/pipx/commands/inject.py [hunk 4]
```diff
@@ -55,6 +55,19 @@ def inject_dep(
             pip_args=pip_args,
             verbose=verbose,
         )
+
+    if not force and venv.has_package(package_name):
+        print(
+            pipx_wrap(
+                f"""
+                {package_name} already seems to be injected in {venv.name!r}.
```
Comment (chrysle) [inline src/pipx/commands/inject.py:63]:
```suggestion
                {hazard} {package_name} already seems to be injected in {venv.name!r}.
```

File: src/pipx/commands/inject.py (hunk 2 above)
Comment (huxuan) [reply src/pipx/commands/inject.py:59]: I noticed the `list_installed_packages()` but it is implemented by run a pip command as subprocess, so I did not reuse it.

**Review by chrysle — DISMISSED** [review]

**Comment by chrysle** [comment]
Looks like you need to fix a test..

**Comment by huxuan** [comment]
> Looks like you need to fix a test..

Sorry for the test broken, I should test it locally.

The test case try to inject `isort` into `pylint` [1] but actually `isort` is already a dependency of `pylint` [2], that is why `isort` is not injected. So maybe I should change the test case? Since the injection is meaningless.

[1] https://github.com/pypa/pipx/blob/ac70dc0d2221bc6a78e39bef86f43b5c73ef03d7/tests/test_list.py#L97
[2] https://github.com/pylint-dev/pylint/blob/10a093ea85f8cb2a3268f439c3b1f40319023956/pyproject.toml#L45

**Comment by huxuan** [comment]
Hi @chrysle, I fixed the test by changing `isort` to `black`, all local tests passed. PTAL :-)

image

**Review by chrysle — APPROVED** [review]

**Comment by chrysle** [comment]
Thank you!

**Pull request merged**